lib/circuit.js

```diff
@@ -138,7 +154,16 @@ class CircuitBreaker extends EventEmitter {
     this[VOLUME_THRESHOLD] = Number.isInteger(options.volumeThreshold)
       ? options.volumeThreshold : 0;
     this[WARMING_UP] = options.allowWarmUp === true;
-    this[STATUS] = new Status(this.options);
+
+    // The user can pass in a Status object ot inialize the Status/stats
+    if (this.options.status && this.options.status instanceof Status) {
+      // Do a check that this is a Status Object,
+      // should it error if it is not a Status Object
```

Comment:
I'm leaning towards yes. It seems wrong to just accept bad input and not do anything about it.

Reply:
same,  i'll update to throw an error if the status option is not an instance of the Status class

Reply:
updated